graph/src/util/lfu_cache.rs

```diff
@@ -79,7 +85,7 @@ impl<K: Clone + Ord + Eq + Hash + Debug, V: CacheWeight + Default> LfuCache<K, V
 
     /// Updates and bumps freceny if already present.
     pub fn insert(&mut self, key: K, value: V) {
-        let weight = value.weight();
+        let weight = CacheEntry::weight(&key, &value);
```

Comment:
This is an existing problem so there is no regression, but the weight of the LfuCache itself (in particular the `queue` field) may not be insignificant for small K/V.

Reply:
Yeah, there's still quite a bit of wiggle room - if what I have so far makes the total cache size more predictable, I'll stick with it; otherwise, we'll need another round of improving estimates. It's crazy how something that would just be an exercise in pointer chasing in C is so hard here (though it would be similarly hard in any language that hides implementation details, like C++)